### PR TITLE
chore(Dockerfile): update docker image

### PR DESCRIPTION
**Pull request by arriqaaq**
- build from go 1.18 base image
- build from distroless image

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -38,3 +47,67 @@ jobs:
...
+        JOB_NAME: ${{ github.job }}
+        JOB_ID: ${{ github.run_id }}
+        DOCKER_IMAGE_IMMUGW: "codenotary/immugw"
+        DOCKER_BUILDKIT: "1"
+      steps:
+        - uses: actions/checkout@v3
+        - name: Build docker images
+          shell: bash
+          run: |
+            if [[ "${GITHUB_REF}" =~ refs/tags/v([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
```
Comment (byo) [inline .github/workflows/push.yml:95]: This requires the same fix as in the immudb project:

```suggestion
            if [[ "${GITHUB_REF}" =~ ^refs/tags/v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
```

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -38,3 +47,67 @@ jobs:
...
+              VERSION_TAG="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.${BASH_REMATCH[3]}"
+              VERSION_TAG_SHORT="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"
+            fi
+
+            docker build --tag "${DOCKER_IMAGE_IMMUGW}:dev" -f Dockerfile .
+
+            # docker login -u "${{ secrets.REGISTRY_USER }}" -p "${{ secrets.REGISTRY_PASS }}"
```
Comment (byo) [inline .github/workflows/push.yml:102]: This still contains some test code - I guess it is for some future updates ?

File: Makefile [hunk 3]
```diff
@@ -100,4 +100,8 @@ dist/binary.md:
 		printf "[$$ff](https://github.com/codenotary/immugw/releases/download/v${VERSION}/$$ff) | $$shm_id \n" ; \
 	done
 
+.PHONY: dist
+dist: dist/binaries
```
Comment (byo) [inline Makefile:104]: This is to keep it consistent with immudb Makefile ?

File: .github/workflows/push.yml (hunk 1 above)
Comment (arriqaaq) [reply .github/workflows/push.yml:95]: ack

File: .github/workflows/push.yml (hunk 2 above)
Comment (arriqaaq) [reply .github/workflows/push.yml:102]: my bad, just wanted to test it out

File: Makefile (hunk 3 above)
Comment (arriqaaq) [reply Makefile:104]: ack, just wanted to follow the same convention

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -38,3 +47,67 @@ jobs:
...
+            name: immugw-binaries
+            path: dist
+            retention-days: 5
+        - name: Calculate checksums
+          run: make dist/binary.md
+
+  images:
+      name: Build and notarize Docker Images
+      needs:
+        - binaries
```
Comment (byo) [inline .github/workflows/push.yml:81]: I think those are not valid dependencies

Comment (arriqaaq) [reply .github/workflows/push.yml:84]: ack, removed them

**Review by byo — APPROVED** [review]

**Pull request merged**